File: app/src/main/java/android/itacademy/by/menu/MenuActivity.java
package android.itacademy.by.menu;

import android.app.Activity;
import android.content.Intent;
import android.itacademy.by.dz1.FirstActivity;
import android.itacademy.by.dz2.FlagsActivity;
import android.itacademy.by.dz3.ImageActivity;
import android.itacademy.by.dz3.LoginActivity;
import android.itacademy.by.dz4.ClockActivity;
import android.itacademy.by.dz4.SovaActivity;
import android.itacademy.by.dz6.activities.MainActivity;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.Button;

public class MenuActivity extends Activity implements View.OnClickListener {
    private int[] diagNums = {10, 5, 3};

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.menu);
        Button buttonDz1 = findViewById(R.id.buttonDz1);
        buttonDz1.setOnClickListener(this);
        Button buttonDz2 = findViewById(R.id.buttonDz2);
        buttonDz2.setOnClickListener(this);
        Button buttonLogin = findViewById(R.id.buttonLogin);
        buttonLogin.setOnClickListener(this);
        Button buttonDz3 = findViewById(R.id.dz3);
        buttonDz3.setOnClickListener(this);
        Button buttonDz4 = findViewById(R.id.dz4);
        buttonDz4.setOnClickListener(this);
        Button sova = findViewById(R.id.sovaButton);
        sova.setOnClickListener(this);
        Button dz5Button = findViewById(R.id.dz5Button);
        dz5Button.setOnClickListener(this);
        Button dz6Button = findViewById(R.id.dz6Button);
        dz6Button.setOnClickListener(this);

    }

    @Override
    public void onClick(View v) {

        switch (v.getId()) {
            case R.id.buttonDz1: {
                Intent intent = new Intent(this, FirstActivity.class);
                startActivity(intent);
                break;
            }
            case R.id.buttonDz2: {
                Intent intent2 = new Intent(this, FlagsActivity.class);
                startActivity(intent2);
                break;
            }
            case R.id.buttonLogin:
                Intent intent3 = new Intent(this, LoginActivity.class);
                startActivity(intent3);
                break;

            case R.id.dz3:
                Intent intent4 = new Intent(this, ImageActivity.class);
                startActivity(intent4);
                break;
            case R.id.dz4:
                Intent intent5 = new Intent(this, ClockActivity.class);
                startActivity(intent5);
                overridePendingTransition(R.anim.entering, R.anim.exiting);
                break;
            case R.id.sovaButton:
                Intent intent6 = new Intent(this, SovaActivity.class);
                startActivity(intent6);
                overridePendingTransition(R.anim.entering, R.anim.exiting);
                break;
            case R.id.dz6Button:
                Intent intent8 = new Intent(this, MainActivity.class);
                startActivity(intent8);
                overridePendingTransition(R.anim.entering,R.anim.exiting);
                break;
        }
    }
}

File: app/src/main/java/android/itacademy/by/menu/MenuActivity.kt
package android.itacademy.by.menu

import android.app.Activity
import android.content.Intent
import android.itacademy.by.dz1.FirstActivity
import android.itacademy.by.dz14.activities.Dz14Activity
import android.itacademy.by.dz2.FlagsActivity
import android.itacademy.by.dz3.ImageActivity
import android.itacademy.by.dz3.LoginActivity
import android.itacademy.by.dz4.ClockActivity
import android.itacademy.by.dz4.SovaActivity
import android.itacademy.by.dz6.activities.MainActivity
import android.os.Bundle
import android.view.View
import android.widget.Button

class MenuActivity : Activity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.menu)
        val buttonDz1 = findViewById<Button>(R.id.buttonDz1)
        buttonDz1.setOnClickListener(this)
        val buttonDz2 = findViewById<Button>(R.id.buttonDz2)
        buttonDz2.setOnClickListener(this)
        val buttonLogin = findViewById<Button>(R.id.buttonLogin)
        buttonLogin.setOnClickListener(this)
        val buttonDz3 = findViewById<Button>(R.id.dz3)
        buttonDz3.setOnClickListener(this)
        val buttonDz4 = findViewById<Button>(R.id.dz4)
        buttonDz4.setOnClickListener(this)
        val sova = findViewById<Button>(R.id.sovaButton)
        sova.setOnClickListener(this)
        val dz5Button = findViewById<Button>(R.id.dz5Button)
        dz5Button.setOnClickListener(this)
        val dz6Button = findViewById<Button>(R.id.dz6Button)
        dz6Button.setOnClickListener(this)
        val dz7Button = findViewById<Button>(R.id.dz7Button)
        dz7Button.setOnClickListener(this)
    }

    override fun onClick(v: View) {

        when (v.id) {
            R.id.buttonDz1 -> {
                val intent = Intent(this, FirstActivity::class.java)
                startActivity(intent)
            }
            R.id.buttonDz2 -> {
                val intent2 = Intent(this, FlagsActivity::class.java)
                startActivity(intent2)
            }
            R.id.buttonLogin -> {
                val intent3 = Intent(this, LoginActivity::class.java)
                startActivity(intent3)
            }

            R.id.dz3 -> {
                val intent4 = Intent(this, ImageActivity::class.java)
                startActivity(intent4)
            }
            R.id.dz4 -> {
                val intent5 = Intent(this, ClockActivity::class.java)
                startActivity(intent5)
                overridePendingTransition(R.anim.entering, R.anim.exiting)
            }
            R.id.sovaButton -> {
                val intent6 = Intent(this, SovaActivity::class.java)
                startActivity(intent6)
                overridePendingTransition(R.anim.entering, R.anim.exiting)
            }
            R.id.dz6Button -> {
                val intent8 = Intent(this, MainActivity::class.java)
                startActivity(intent8)
                overridePendingTransition(R.anim.entering, R.anim.exiting)
            }
            R.id.dz7Button -> {
                val intent9 = Intent(this, Dz14Activity::class.java)
                startActivity(intent9)
                overridePendingTransition(R.anim.entering, R.anim.exiting)
            }
        }
    }
}